src/services/asset-movement/common.ts

```diff
@@ -315,21 +511,269 @@ export type KeetaAssetMovementAnchorGetTransferStatusResponse = ({
 	error: string;
 });
 
-export type KeetaAssetMovementAnchorCreatePersistentForwardingRequest = {
+type PhysicalAddress = {
+	line1: string;
+	line2?: string;
+	country: ISOCountryCode;
+	postalCode: string;
+	subdivision: string;
+	city: string;
+};
+
+type USBankAccountType = 'checking' | 'savings';
+
+export type BankAccountAddressResolved = {
+	type: 'bank-account';
+	accountAddress?: PhysicalAddress | string;
+	obfuscated?: false;
+
+	bankName?: string;
+
+	accountOwner: {
+		type: 'individual';
+		firstName: string;
+		lastName: string;
+	} | {
+		type: 'business';
+		businessName: string;
+	} | {
+		type: 'unknown';
+		beneficiaryName: string;
+	}
+} & ({
+	accountType: 'us';
+
+	accountNumber: string;
+	routingNumber: string;
+	accountTypeDetail: USBankAccountType;
+} | {
+	accountType: 'iban-swift';
+
+
+	country?: ISOCountryCode;
+
+	accountNumber: string;
+	bic?: string;
+
+	iban?: string;
+
+	bankAddress?: PhysicalAddress;
+
+	swift?: {
+		category: string;
+		purposeOfFunds: string[];
+		businessDescription: string;
+	}
+} | {
+	accountType: 'clabe';
+
+	accountNumber: string;
+} | ({
+	accountType: 'pix';
+	document?: {
+		type?: 'cpf' | 'cnpj';
+		number: string;
+	}
+} & ({
+	brCode: string;
+} | {
+	pixKey: string;
+})));
+
+export type BankAccountAddressObfuscated = {
+	type: 'bank-account';
+	obfuscated: true;
+
+	accountOwner?: {
+		type?: 'individual' | 'business';
+		name?: string;
+		businessName?: string;
+	}
+
+	bankName?: string;
+
+	accountNumberEnding?: string;
+} & ({
+	accountType: 'us';
+
+	routingNumber: string;
+	accountTypeDetail?: USBankAccountType;
+
+} | {
+	accountType: 'iban-swift';
+	country?: ISOCountryCode;
+	bic?: string;
+} | {
+	accountType: 'clabe';
+} | {
+	accountType: 'pix';
+})
+
+type CryptoAddress = string;
+type AddressResolved = BankAccountAddressResolved | CryptoAddress;
+type AddressObfuscated = BankAccountAddressObfuscated | CryptoAddress;
+
+export type PersistentAddressTemplateData = {
+	id: string;
+	location: AssetLocationLike;
 	asset: MovableAsset;
-	destinationLocation: AssetLocationLike;
-	destinationAddress: string;
-	sourceLocation: AssetLocationLike;
+	address: AddressObfuscated;
 }
 
-export type KeetaAssetMovementAnchorCreatePersistentForwardingResponse = ({
+export type KeetaAssetMovementAnchorCreatePersistentForwardingAddressTemplateClientRequest = {
+	account?: KeetaNetAccount;
+	asset: MovableAsset;
+	location: AssetLocationLike;
+	address: AddressResolved;
+}
+
+
+export type KeetaAssetMovementAnchorCreatePersistentForwardingAddressTemplateRequest = ToJSONSerializable<Omit<KeetaAssetMovementAnchorCreatePersistentForwardingAddressTemplateClientRequest, 'asset' | 'location' | 'address'>> & {
```

Comment:
Rather than repeating this pattern over and over it might be useful to create a utility type

```typescript
type ConvertToExternalRequest<Internal extends object, Overrides extends object, Signed extends boolean = false> = ToJSONSerializable<Omit<Internal, keyof Overrides>> & Overrides & Signed extends true ? { signed?: HTTPSignedField | undefined } : {};